numpy/_core/tests/test_datetime.py

```diff
@@ -2552,6 +2558,80 @@ def test_limit_str_roundtrip(self, time_unit, sign):
         limit_via_str = np.datetime64(str(limit), time_unit)
         assert limit_via_str == limit
 
+    def test_datetime_hash(self):
```

Comment:
not sure if numpy does things differently from pandas, but id suggest splitting this into targeted tests with comments about which invariance is being tested in each place

Reply:
The numpy tests tend to be quite coarse-grained.

Reply:
Old ones, we/I prefer fine grained for various reasons.  But I don't care enough here to nitpick about it this time around.

Reply:
I made the tests more fine-grained.